arangosh/Benchmark/BenchFeature.cpp

```diff
@@ -333,19 +399,15 @@ void BenchFeature::start() {
   // aggregated stats for all runs
   BenchmarkStats totalStats;
 
-  auto builder = std::make_shared<VPackBuilder>();
-  builder->openObject();
-  builder->add("histogram", VPackValue(VPackValueType::Object));
+  VPackBuilder builder;
+  builder.openObject();
   std::vector<std::unique_ptr<BenchmarkThread>> threads;
   bool ok = true;
   std::vector<BenchRunResult> results;
   std::stringstream pp;
-  pp << "Interval/Percentile:";
-  for (auto percentile : _percentiles) {
-    pp << std::fixed << std::right << std::setw(12) << std::setprecision(2)
-       << percentile << "%";
+  if (_generateHistogram) {
+    setupHistogram(pp, builder);
```

Comment:
it's really hard to follow who is responsible for forming a builder structure. 
I'd be in favor of at least moving `builder.add("histogram", VPackValue(VPackValueType::Object));` out of `setupHistogram`, so everything is handled in `BenchFeature::start()`

Reply:
Addressed in latest commit